src/vos/pmdk_log.c

```diff
@@ -83,8 +60,33 @@ pmdk_log_function(enum pmemobj_log_level level, const char *file_name, unsigned
  * A custom variant of _D_LOG_NOCHECK() which passes the file name,
  * line number and the function name from the local variables.
  */
+#define PMDK_LOG_FUNCTION_MAX_FILENAME 255
 #define PMDK_LOG_NOCHECK(mask, fmt, ...)                                                           \
-	d_log(mask, "%s:%d %s() " fmt, file_name_buff, line_no, function_name, ##__VA_ARGS__)
+	do {                                                                                       \
+		/* normalize file path - remove leading "../" */                                   \
+		while ((*file_name == '.') && (*(file_name + 1) == '.') &&                         \
+		       (*(file_name + 2) == '/')) {                                                \
+			file_name += 3;                                                            \
+		}                                                                                  \
+                                                                                                   \
+		/* simplify file path by removing cyclic pattern "src/../src" */                   \
+		if (strncmp(file_name, "src/../src", sizeof("src/../src") - 1) == 0) {             \
+			file_name += sizeof("src/../") - 1;                                        \
+		}                                                                                  \
+		/* Add "pmdk/" prefix to file name */                                              \
+		/* Prefix is needed to filter out PMDK messages in NLT results analysis */         \
+		/* as it is implemented in https://github.com/daos-stack/pipeline-lib/pull/457 */  \
+                                                                                                   \
+		char  file_name_buff[PMDK_LOG_FUNCTION_MAX_FILENAME] = "pmdk/";                    \
```

Comment:
minor nit, we usually put declarations at the top of the current block.

Reply:
Fixed in: https://github.com/daos-stack/daos/pull/14923/commits/4f2622c6ad648d44393b47f4cfb1c7322ce394f7